woocommerce/woocommerce
# Determine _product_template_id from WP filter

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This creates a filter that allows extensions to determine a product template ID when it is not defined. This is to allow products created in the classic editor to match correctly to a product template depending on characteristics of the product.

PS: There is also an unrelated tweak to an E2E test made by @lanej0 that was causing the build to fail.









### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Install and activate the following plugin: [test-match-product-template.zip](https://github.com/user-attachments/files/15660578/test-match-product-template.zip)

2. DISABLE the new product editor at /wp-admin/admin.php?page=wc-settings&tab=advanced&section=features
3. Go to Products > Add
4. Create a simple product and add "15" to the regular price, publish it
5. Enable the new product editor at /wp-admin/admin.php?page=wc-settings&tab=advanced&section=features
6. With the network tab opened, open the product just created
7. Check the product endpoint response (`/wp-json/wc/v3/products/[PRODUCT_ID]`), it should contain a `meta_data` entry with key `_product_template_id` and value `test-specific-product-template`.
8. Disable the new product editor
9. Open the same product with the classic editor
10. Make any change and save it
11. Browse your database, run the following query with the just saved product id: `SELECT * FROM wp_postmeta WHERE post_id = <product_id>`
12. There should not be any entry with `meta_key` equal to `_product_template_id`, as it was deleted.
 




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
